File: katas/5 kyu/not-very-secure/solution.py
import codewars_test as test
import re
# -------------------------------- my solution ------------------------------- #
def alphanumeric(password):
    regex = '^(?=[^\s_])[a-zA-Z0-9]+$'
    pattern = re.compile(regex)
    return bool(pattern.match(password))
# ------------------------------ clever solution ----------------------------- #
def alphanumeric(string):
    return string.isalnum()
# ----------------------------------- test ----------------------------------- #
@test.describe("Sample Tests")
def sample_tests():
    tests = [
        ("hello world_", False),
        ("PassW0rd", True),
        ("     ", False)
    ]
    for s, b in tests:
        @test.it('alphanumeric("' + s + '")')
        def sample_test():
            test.assert_equals(alphanumeric(s), b)

File: katas/5 kyu/not-very-secure/solution.php
<!-- not-very-secure -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function alphanumeric(string $s): bool {
  $regex = '/^(?=[^\s_])[a-zA-Z0-9]+$/';
  return preg_match_all($regex , $s , $matches);
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function alphanumeric(string $s): bool {
  return ctype_alnum($s);
}
/* ---------------------------------- TEST ---------------------------------- */
class AlphanumericTest extends TestCase {
  public function testExamples() {
    $this->assertTrue(alphanumeric('Mazinkaiser'));
    $this->assertFalse(alphanumeric('hello world_'));
    $this->assertTrue(alphanumeric('PassW0rd'));
    $this->assertFalse(alphanumeric('     '));
  }
}